themes/default/content/blog/managing-github-with-pulumi/index.md

```diff
@@ -0,0 +1,599 @@
+
+1. Create a parent team, using a team name
+2. Obtain its team ID
+3. Create any child teams, also using a team name
+4. Write the parent team’s ID into the `ParentTeamId field` of each child team.
+5. Do all of the above in a single `pulumi up`.
+
+This is where maintaining infrastructure with Pulumi truly shines. In our code, we can use [Pulumi Apply](https://www.pulumi.com/docs/intro/concepts/inputs-outputs/#apply) to hold on to the promise of a parent team ID, and pass this promise into the appropriate field:
+
+```go
+func setupTeams(ctx *pulumi.Context, parentTeam *Team) error {
+
+// set up parent team, i.e. engineering
+  ghParentTeam, err := github.NewTeam(ctx, parentTeam.Slug, &github.TeamArgs{
+     Description: pulumi.String(parentTeam.Description),
+     Name:        pulumi.String(parentTeam.Name),
+     Privacy:     pulumi.String("closed"),
+  }, pulumi.Protect(false))
+  if err != nil {
+     fmt.Println("encountered error creating new Pulumi github team: ", parentTeam.Name)
+     return err
+  }
+
+//set up nested teams
+  for _, childTeam := range parentTeam.Teams {
+     // set each child team's parent team ID to the current team ID
+     ghChildTeam, err := github.NewTeam(ctx, childTeam.Slug, &github.TeamArgs{
+        Description: pulumi.String(childTeam.Description),
+        Name:        pulumi.String(childTeam.Name),
+        Privacy:     pulumi.String("closed"),
+        ParentTeamId: ghParentTeam.ID().ApplyT(func(id interface{}) int {
+        // we need to re-cast id as an int so we can then transform it into a pulumi.IntOutput, which can be used to set the ParentTeamId.
+           x := fmt.Sprintf("%v", id)
+           y, _ := strconv.Atoi(x)
+           return y
+        }).(pulumi.IntOutput),
+     }, pulumi.Protect(false))
+     if err != nil {
+        fmt.Println("encountered error creating new Pulumi github team: ", childTeam.Name)
+        return err
+     }
+  }
+  return nil
+}
+```
+
+Running this as part of `main.go` will result in beautifully nested teams on the GitHub UI. But with Pulumi, we can do even better. We can set [`pulumi.Parent()`](https://www.pulumi.com/docs/intro/concepts/resources/options/parent/) on the child teams:
+
+```go
+for _, childTeam := range parentTeam.Teams {
+    ghChildTeam, err := github.NewTeam(ctx, childTeam.Slug, &github.TeamArgs{#...},
+    pulumi.Protect(false),
+    pulumi.Parent(ghParentTeam))
+    #...
+}
+```
+
+With this, we can see the relationship between the resources on `pulumi up`:
+
+```bash
+Updating (prod)
+
+     Type                     Name            Status
+     pulumi:pulumi:Stack      team-mgmt-prod
+ +   └─ github:index:Team     newteam         created
+ +      └─ github:index:Team  newchildteam    created
+Resources:
+    + 2 created
+2 unchanged
+```
+
+Which will be reflected on the Pulumi stack’s Graph View in the Pulumi Service:
```

Comment:
Stack is lowercase here, but it's been uppercase above. I could go either way as long as it's consistent.